src/thunderbots/software/test/ai/passing/pass_generator.cpp

```diff
@@ -29,54 +75,159 @@ class PassGeneratorTest : public testing::Test
-               Timestamp::fromSeconds(0)),
-         Robot(3, {-1, -1}, {-0.5, 0}, Angle::zero(), AngularVelocity::zero(),
-               Timestamp::fromSeconds(0)),
-         Robot(4, {0.2, 0.5}, {-0.5, 0}, Angle::zero(), AngularVelocity::zero(),
-               Timestamp::fromSeconds(0))});
+    friendly_team.updateRobots({
+        Robot(3, {-1, -1}, {-0.5, 0}, Angle::zero(), AngularVelocity::zero(),
+              Timestamp::fromSeconds(0)),
+    });
     world.updateFriendlyTeamState(friendly_team);
     Team enemy_team(Duration::fromSeconds(10));
-    enemy_team.updateRobots({Robot(0, {1, 0}, {0, 0}, Angle::zero(),
-                                   AngularVelocity::zero(), Timestamp::fromSeconds(0)),
-                             Robot(1, {2, 1}, {-0.5, 0}, Angle::zero(),
+    enemy_team.updateRobots({Robot(0, {0.5, 0.5}, {-0.5, 0}, Angle::zero(),
                                    AngularVelocity::zero(), Timestamp::fromSeconds(0)),
-                             Robot(2, {3, 2}, {-0.5, 0}, Angle::zero(),
+                             Robot(1, {-2, -2}, {-0.5, 0}, Angle::zero(),
                                    AngularVelocity::zero(), Timestamp::fromSeconds(0)),
-                             Robot(3, {4, -1}, {-0.5, 0}, Angle::zero(),
+                             Robot(2, {3, -2}, {-0.5, 0}, Angle::zero(),
                                    AngularVelocity::zero(), Timestamp::fromSeconds(0)),
-                             Robot(4, {0.5, 4}, {-0.5, 0}, Angle::zero(),
+                             Robot(3, {-2.5, 4}, {-0.5, 0}, Angle::zero(),
                                    AngularVelocity::zero(), Timestamp::fromSeconds(0))});
     world.updateEnemyTeamState(enemy_team);
 
     pass_generator->setWorld(world);
 
-    std::this_thread::sleep_for(10s);
+    // Wait until the pass stops improving or 30 seconds, whichever comes first
+    waitForConvergence(pass_generator, 0.01, 30);
 
     // Find what pass we converged to
-    auto converged_pass_opt = pass_generator->getBestPassSoFar();
-    ASSERT_TRUE(converged_pass_opt);
+    auto [converged_pass, converged_score] = pass_generator->getBestPassSoFar();
 
     // Check that we keep converging to the same pass
     for (int i = 0; i < 7; i++)
     {
         std::this_thread::sleep_for(0.5s);
-        auto new_pass_opt = pass_generator->getBestPassSoFar();
-        ASSERT_TRUE(new_pass_opt);
-
-        EXPECT_EQ(converged_pass_opt->passerPoint(), new_pass_opt->passerPoint());
-        EXPECT_LE(
-            (converged_pass_opt->receiverPoint() - new_pass_opt->receiverPoint()).len(),
-            0.2);
-        EXPECT_LE(converged_pass_opt->speed() - new_pass_opt->speed(), 0.2);
-        EXPECT_LE(abs((converged_pass_opt->startTime() - new_pass_opt->startTime())
-                          .getSeconds()),
-                  0.2);
+        auto [pass, score] = pass_generator->getBestPassSoFar();
+
+        std::cout << converged_pass << std::endl;
```

Comment:
Probably don't want printouts in the test

Reply:
See above comment, but yep, please leave unresolved for now.